### PR TITLE
[1.8.x] fix for mapping error status code for return `nil` type in Ballerina to OpenAPI spec generation

### PR DESCRIPTION
**Pull request by lnash94**
## Automation tests
 - Unit tests - Added
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: openapi-cli/src/test/resources/ballerina-to-openapi/response/nil_return_type.bal [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    resource function head path1() {
+    };
+    resource function head path_with_query(string id)  {
+    };
+    resource function head path_with_path/[string id]() {
+    };
+    resource function head path_with_headers(@http:Header string header) {
+    };
+    // resource function head path_with_request_body(@http:Payload string payload) {
+    // };
```
Comment (NipunaRanasinghe) [inline openapi-cli/src/test/resources/ballerina-to-openapi/response/nil_return_type.bal:68]: Any reason to disable this request body scenario?

Comment (lnash94) [reply openapi-cli/src/test/resources/ballerina-to-openapi/response/nil_return_type.bal]: Yeah, these http methods `HEAD` and `GET` aren't allowed to have a request body entity payload, therefore these scenarios capture within negative tests https://github.com/ballerina-platform/openapi-tools/pull/1535/commits/2224d4ac06d6b0f8a6e60c6785bf04d520296a06.

Comment (NipunaRanasinghe) [reply openapi-cli/src/test/resources/ballerina-to-openapi/response/nil_return_type.bal]: Got it. In that case we don't need to keep the commented code right?

Comment (lnash94) [reply openapi-cli/src/test/resources/ballerina-to-openapi/response/nil_return_type.bal]: Yeah, we removed it :)

File: openapi-cli/src/test/resources/ballerina-to-openapi/response/nil_union_return_type.bal [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+ |};
+
+
+service /payloadV on new http:Listener(9090) {
+    # for `get` method with nil return type and error
+    resource function get path_get_error() returns error? {
+    };
+
+    # for `get` method with nil return type and success status code
+    resource function get path_get() returns string? {
```
Comment (dilanSachi) [inline openapi-cli/src/test/resources/ballerina-to-openapi/response/nil_union_return_type.bal:20]:
```suggestion
    resource function get path_get() returns string {
```
I guess we already have test cases for above right?

Comment (lnash94) [reply openapi-cli/src/test/resources/ballerina-to-openapi/response/nil_union_return_type.bal:20]: Actually no,  we haven't this is regarding the path without any parameters that have returned with string and nil value, in line 24 we have a different scenario that returns the same with the query parameter :)

**Review by dilanSachi — APPROVED** [review]
LGTM

**Pull request merged**